packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_flutter_web.dart

```diff
@@ -0,0 +1,280 @@
+    MarkerId markerId, {
+    @required int mapId,
+  }) async {
+    _map(mapId).showInfoWindow(markerId);
+  }
+
+  /// Hides the [InfoWindow] (if any) of the [Marker] identified by `markerId`.
+  @override
+  Future<void> hideMarkerInfoWindow(
+    MarkerId markerId, {
+    @required int mapId,
+  }) async {
+    _map(mapId).hideInfoWindow(markerId);
+  }
+
+  /// Returns true if the [InfoWindow] of the [Marker] identified by `markerId` is shown.
+  @override
+  Future<bool> isMarkerInfoWindowShown(
+    MarkerId markerId, {
+    @required int mapId,
+  }) async {
+    return _map(mapId).isInfoWindowShown(markerId);
+  }
+
+  /// Returns the zoom level of the `mapId`.
+  @override
+  Future<double> getZoomLevel({
+    @required int mapId,
+  }) {
+    return _map(mapId).getZoomLevel();
+  }
+
+  // The following are the 11 possible streams of data from the native side
+  // into the plugin
+
+  @override
+  Stream<CameraMoveStartedEvent> onCameraMoveStarted({@required int mapId}) {
+    return _events(mapId).whereType<CameraMoveStartedEvent>();
+  }
+
+  @override
+  Stream<CameraMoveEvent> onCameraMove({@required int mapId}) {
+    return _events(mapId).whereType<CameraMoveEvent>();
+  }
+
+  @override
+  Stream<CameraIdleEvent> onCameraIdle({@required int mapId}) {
+    return _events(mapId).whereType<CameraIdleEvent>();
+  }
+
+  @override
+  Stream<MarkerTapEvent> onMarkerTap({@required int mapId}) {
+    return _events(mapId).whereType<MarkerTapEvent>();
+  }
+
+  @override
+  Stream<InfoWindowTapEvent> onInfoWindowTap({@required int mapId}) {
+    return _events(mapId).whereType<InfoWindowTapEvent>();
+  }
+
+  @override
+  Stream<MarkerDragEndEvent> onMarkerDragEnd({@required int mapId}) {
+    return _events(mapId).whereType<MarkerDragEndEvent>();
+  }
+
+  @override
+  Stream<PolylineTapEvent> onPolylineTap({@required int mapId}) {
+    return _events(mapId).whereType<PolylineTapEvent>();
+  }
+
+  @override
+  Stream<PolygonTapEvent> onPolygonTap({@required int mapId}) {
+    return _events(mapId).whereType<PolygonTapEvent>();
+  }
+
+  @override
+  Stream<CircleTapEvent> onCircleTap({@required int mapId}) {
+    return _events(mapId).whereType<CircleTapEvent>();
+  }
+
+  @override
+  Stream<MapTapEvent> onTap({@required int mapId}) {
+    return _events(mapId).whereType<MapTapEvent>();
+  }
+
+  @override
+  Stream<MapLongPressEvent> onLongPress({@required int mapId}) {
+    return _events(mapId).whereType<MapLongPressEvent>();
+  }
+
+  /// Disposes of the current map. It can't be used afterwards!
+  @override
+  void dispose({@required int mapId}) {
+    _map(mapId)?.dispose();
+    _mapById.remove(mapId);
+  }
+
+  @override
+  Widget buildView(
+      Map<String, dynamic> creationParams,
+      Set<Factory<OneSequenceGestureRecognizer>> gestureRecognizers,
+      PlatformViewCreatedCallback onPlatformViewCreated) {
+    int mapId = creationParams.remove('_webOnlyMapCreationId');
+
+    if (mapId == null) {
```

Comment:
Should we just assert `creationParams.containsKey` instead?

Reply:
This should *never* trigger now that `mapId` is auto-generated, so an assert should be enough, yes. This is a left-over from when I was adding the `_webOnlyMapCreationId` parameter.